pkg/kubefed2/options/options.go

```diff
@@ -71,3 +74,31 @@ func (o *CommonSubcommandOptions) SetName(args []string) error {
 	o.ClusterName = args[0]
 	return nil
 }
+
+type FederationConfigOptions struct {
+	ClusterNamespace string
+}
+
+func GetOptionsByFederationConfig(hostConfig *rest.Config, namespace string) (limitedScope bool, clusterNamespace string, err error) {
```

Comment:
`s/GetOptionsByFederationConfig/GetOptionsFromFederationConfig/`

Reply:
Also, please prefer returning variables explicitly setting values defined in the signature.  As per the logical error noted below with `errors.Wrapf`, there is less room for error when returning explicitly.

Reply:
@marun, what did you mean about `explicitly setting values defined in the signature`? I set each returning varible explicitely.

Reply:
My intent was not to have you remove the error wrapping:

```golang
// This is what appeared, and the result is thrown away:
errors.Wrapf(err, "...")
// This is what was likely intended
err = errors.Wrapf(err, "...")
```

Notice, however, that almost no code in this repo uses 'named return values' as this function does.  For consistency, please rewrite this function to use regular unnamed return values.

Reply:
Got it. Addressed.